crypto/evp_extra/p_pqdsa_asn1.c

```diff
@@ -149,7 +149,40 @@ static int pqdsa_priv_decode(EVP_PKEY *out, CBS *params, CBS *key, CBS *pubkey)
     OPENSSL_PUT_ERROR(EVP, EVP_R_DECODE_ERROR);
     return 0;
   }
-  return PQDSA_KEY_set_raw_private_key(out->pkey.pqdsa_key,CBS_data(key));
+
+  // Set the private key
+  if (!PQDSA_KEY_set_raw_private_key(out->pkey.pqdsa_key, CBS_data(key))) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_DECODE_ERROR);
+    return 0;
+  }
+
+  // Calculate public key from private key if method exists
+  if (out->pkey.pqdsa_key->pqdsa->method->pqdsa_pack_pk_from_sk != NULL) {
```

Comment:
Shouldn't we always return failure (`0`) if we're unable to initialize the public key? I assume the `pqdsa_pack_pk_from_sk` function ptr should always be non-NULL here (line 360) so this condition would never fail?

Upon seeing a successful return code, a caller assumes the EVP_PKEY is initialized and might subsequently SEGFAULT on a call to (e.g.) `EVP_PKEY_cmp`.

Reply:
Ah I see. I was trying to prevent against a failure if the ptr is non-null, as you say. Removed this check in [cd8da04](https://github.com/aws/aws-lc/pull/2145/commits/cd8da04f33b1baacbe7857532ea260b3c1669bc3) and cleaned up the surrounding logic to only return `SUCESS` when both priv and pub are populated.